superset/assets/javascripts/SqlLab/components/VisualizeModal.jsx

```diff
@@ -91,16 +89,30 @@ class VisualizeModal extends React.PureComponent {
     });
     if (this.state.chartType === null) {
       hints.push(VISUALIZE_VALIDATION_ERRORS.REQUIRE_CHART_TYPE);
-    } else if (this.state.chartType.requiresTime) {
-      let hasTime = false;
-      for (const colName in cols) {
-        const col = cols[colName];
-        if (col.hasOwnProperty('is_date') && col.is_date) {
-          hasTime = true;
+    } else {
+      if (this.state.chartType.requiresTime) {
+        let hasTime = false;
```

Comment:
i think we could simplify this block by using [Array#some](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some?v=example)

```js
const hasTime = cols.some(col => col.is_date);
if (!hasTime) {
  hints.push(VISUALIZE_VALIDATION_ERRORS.REQUIRE_TIME);
}
```